src/duckling/game/drawable/container-drawable.component.ts

```diff
@@ -0,0 +1,139 @@
+            (addClicked)="onNewDrawableClicked($event)">
+        </dk-enum-choice>
+
+        <div class="drawables-card">
+            <div *ngFor="let key of keys()">
+                <dk-card-list-element
+                    [title]="containerDrawable?.drawables[key].key"
+                    [opened]="cardsOpen[key]"
+                    [first]="firstChildDrawable(key)"
+                    [last]="lastChildDrawable(key)"
+                    (deleted)="onChildCardDeleted(key, $event)"
+                    (toggled)="onChildCardToggled(key, $event)"
+                    (moved)="onChildCardMoved(key, $event)">
+                    <dk-drawable-component
+                        [drawable]="containerDrawable?.drawables[key]"
+                        (drawableChanged)="onChildDrawableChanged(key, $event)">
+                    </dk-drawable-component>
+                </dk-card-list-element>
+                <md-divider *ngIf="lastChildDrawable(key)"></md-divider>
+            </div>
+        </div>
+    `
+})
+export class ContainerDrawableComponent {
+    // hoist DrawableType so template can access it
+    DrawableType = DrawableType;
+
+    @Input() containerDrawable : ContainerDrawable;
+    @Output() drawableChanged = new EventEmitter<ContainerDrawable>();
+    cardsOpen : boolean[] = [];
+
+    onChildDrawableChanged(index : number, newDrawable : Drawable) {
+        var patch = this.containerDrawable.drawables;
+        patch[index] = newDrawable;
+        this.drawableChanged.emit(immutableAssign(this.containerDrawable, {drawables: patch}));
+    }
+
+    onChildCardDeleted(index : number, deleted : boolean) {
+        if (!deleted) {
+            return;
+        }
+        var head = this.containerDrawable.drawables.slice(0, index);
+        var tail = this.containerDrawable.drawables.slice(index + 1, this.containerDrawable.drawables.length);
+        var patch : Drawable[] = head.concat(tail);
+        this.drawableChanged.emit(immutableAssign(this.containerDrawable, {drawables: patch}));
+    }
+
+    onChildCardToggled(index : number, opened : boolean) {
+        this.cardsOpen[index] = opened;
+    }
+
+    onChildCardMoved(index : number, down : boolean) {
+        var newIndex = index + (down ? 1 : -1);
+        var patch = this.containerDrawable.drawables;
+        var bankedItem = patch[newIndex];
+        patch[newIndex] = patch[index];
+        patch[index] = bankedItem;
+
+        var bankedOpened = this.cardsOpen[newIndex];
+        this.cardsOpen[newIndex] = this.cardsOpen[index];
+        this.cardsOpen[index] = bankedOpened;
+
+        this.drawableChanged.emit(immutableAssign(this.containerDrawable, {drawables: patch}));
+    }
+
+    onNewDrawableClicked(pickedType : DrawableType) {
+        this.drawableChanged.emit(immutableAssign(this.containerDrawable, {
+            drawables: this.containerDrawable.drawables.concat([getDefaultDrawable(pickedType)])
+        }));
+    }
```

Comment:
var newDrawable = immutableAssign(getDefaultDrawable(default), {key: findNextUniqueKey(pickedtype)); 
this.drawableChanged.emit(immutableAssign(this.containerDrawable, {
        drawables: this.containerDrawable.drawables.concat(newDrawable)
}));